lib/search/client/elasticsearch.rb

```diff
@@ -1,77 +1,85 @@
 require "search/client"
+require 'search/client/search_result'
+require 'elasticsearch'
 
 class Search
   class Client
     class Elasticsearch < Search::Client
+      attr_accessor :client, :index_name, :settings, :type
       ESCAPE_LUCENE_CHARS = /
         ( [-+!\(\)\{\}\[\]^"~*?:\\] # A special character
           | &&                        # Boolean &&
           | \|\|                      # Boolean ||
         )/x
 
-      def initialize(config)
-        super
+      def initialize(index_name:, settings:, type:, config:)
+        super(config)
+        @index_name = index_name
+        @settings = settings
+        @type = type
         @config = config
-        configure
-      end
-
-      def configure
-        url = @config[:url]
-        Tire.configure { url url }
-        logger = @config[:logger]
-        if logger
-          Tire.configure { logger logger }
-        end
-      end
-
-      def indexer
-        return @indexer unless @indexer.nil?
-        @indexer = Tire::Index.new(@config[:index])
+        @client = ::Elasticsearch::Client.new(config)
       end
 
       def delete_index
-        indexer.delete
+        if client.indices.exists?(index: index_name)
+          client.indices.delete(index: index_name)
+        end
       end
 
       def pre_index
-        indexer.delete
-        raise unless indexer.create(@config[:create])
+        delete_index
+        index = client.indices.create(
+          index: index_name,
+          body: settings
+        )
+        raise unless index
```

Comment:
I'm not sure. I left it pretty much as it was before mostly for lack of context: https://github.com/alphagov/licence-finder/blob/master/lib/search/client/elasticsearch.rb#L38